frontend/src/components/PrerequisiteTree/PrerequisiteTree.tsx

```diff
@@ -113,15 +113,17 @@ const PrerequisiteTree = ({ courseCode }: Props) => {
 
       // render graph
       if (!graph) {
-        generateTreeGraph(graphData);
+        if (graphData.children.length !== 0) {
+          generateTreeGraph(graphData);
+        } else {
+          <p>No prerequisite visualisation is needed for this course</p>;
```

Comment:
Since this is in the useEffect, it is not returning this html and thus not showing it. You would need to conditionally render this around line ~132
```
{loading && <Spinner text="Loading tree..." /> }
{!loading && <TODO> && <p> No prerequisite visualisation is needed for this course </p> }
```
Replace <TODO> from above with the logic to check if the graph has any prereqs and unlocks nodes. You can't use`graphData` unlike in line 116 right now since that is only the scope of the useeffect call. I think you can check via the `graph` state variable (i.e. get num nodes of graph, if its only one, then its only has no unlocks/prereqs nodes)?

Reply:
Oh you can just console log the graph getters and setters, since the graph var is just an instance of a class. Typescript can just tell you what methods there are to extract the graph internal representation

Reply:
Also, feel free to revert this change since you dont need this logic anymore?